### PR TITLE
Fixed #23727 -- Inhibited the post_migrate signals when using serialized...

### PR DESCRIPTION
**Pull request by tbeadle**
When using a TransactionTestCase-based class with serialized_rollback set to
True, after creating the database and running its migrations (along with
emitting the post_migration signal), the content of the database is serialized
to _test_serialized_contents.  Without this fix, after the first test case,
_fixture_teardown would flush the tables but then the post_migrate signal would be
emitted and new rows (with new PK's) would be created in the django_content_type
table.  Then, when starting the second or any subsequent test cases in a suite,
_fixture_setup attempts to deserialize the content of _test_serialized_contents,
but these rows are identical to the rows already in the database, except for
their PK's.  This causes an IntegrityError due to a uniqueness constraint in the
django_content_type table.
This change made it so that, in the above scenario, the post_migrate signal is
not emitted after flushing the tables, since it will be repopulated during
_fixture_setup.

The problem is simple to reproduce with the following app test code:

    class TestIt(TransactionTestCase):
        serialized_rollback = True
    
       def test_one(self):
           self.assertTrue(True)
    
       def test_two(self):
           self.assertTrue(True)

however this proved to be very difficult to write a test case for in the
django unit tests framework because the framework requires that the test case
class's 'available_apps' attribute be non-None, but doing so would cause the
suppression of the post_migrate signal to be emitted anyway.

### CODE REVIEW

**Comment by MarkusH** [comment]
As far as I understand the code involved in this change, this patch makes sense, though I don't understand your explanation of why it's hard to write a test yet.

Furthermore, I haven't found a place where the `_test_serialized_contents` is ever being unset. Not sure though if this is required/important.

**Comment by timgraham** [comment]
I think we could we set `available_apps = None` to suppress the warning from runtests.py saying "Please define available_apps in TransactionTestCase and its subclasses". Then could we do something like connect a post_migrate signal that raises an exception, thereby testing that it isni't called?

**Comment by timgraham** [comment]
Added a test in https://github.com/django/django/pull/5168.
